File: app/src/main/java/com/jeeps/gamecollector/comparators/GameByTimesPlayedComparator.java
package com.jeeps.gamecollector.comparators;

import com.jeeps.gamecollector.remaster.data.model.data.Game;

import java.util.Comparator;

/**
 * Created by jeeps on 1/9/2018.
 */

public class GameByTimesPlayedComparator implements Comparator<Game> {

    private boolean desc;

    public GameByTimesPlayedComparator(boolean desc) {
        this.desc = desc;
    }

    public GameByTimesPlayedComparator() {
        this.desc = false;
    }

    @Override
    public int compare(Game game1, Game game2) {
        if (!desc)
            return Integer.compare(game1.getTimesCompleted(), game2.getTimesCompleted());
        return Integer.compare(game1.getTimesCompleted(), game2.getTimesCompleted()) * -1;
    }
}

File: app/src/main/java/com/jeeps/gamecollector/remaster/utils/comparators/GameByTimesPlayedComparator.kt
package com.jeeps.gamecollector.remaster.utils.comparators

import com.jeeps.gamecollector.remaster.data.model.data.games.Game

/**
 * Created by jeeps on 1/9/2018.
 */
class GameByTimesPlayedComparator : Comparator<Game> {
    private var desc: Boolean

    constructor(desc: Boolean) {
        this.desc = desc
    }

    constructor() {
        desc = false
    }

    override fun compare(game1: Game, game2: Game): Int {
        return if (!desc)
            game1.timesCompleted.compareTo(game2.timesCompleted)
        else
            game1.timesCompleted.compareTo(game2.timesCompleted) * -1
    }
}